.github/workflows/docs-preview-prepare.yml

```diff
@@ -33,11 +28,21 @@ jobs:
     runs-on: ubuntu-24.04
     steps:
       - uses: actions/checkout@v4
+        with:
+          # As `on: pull_request_target` runs this workflow from the default branch of our repo,
+          # Adjust to checkout the correct PR branch (including repo to support PRs from forks):
+          ref: ${{ env.PR_REF }}
+          repository: ${{ env.PR_REPO }}
+          # Prevent `secrets.GITHUB_TOKEN` from being stored in a `.git/config` file, which could otherwise
+          # be compromised when executing untrusted code from a PR (as is done in the build step below).
+          persist-credentials: false
```

Comment:
Without this, my understanding is the equivalent of `secrets.GITHUB_TOKEN` would be stored in `.git/config`, so whatever `permissions` grant, the PR author could leverage for malicious intent should they gain access to the token.

I'm not sure how many other actions may cache the token, or other secrets by default like `actions/checkout` does implicitly. I don't think we would have many scenarios for workflows requiring to execute untrusted third-party content + access to secrets... presumably this is mostly a concern for PR oriented workflows when `pull_request` isn't sufficient?

Reply:
The token is available to an attacker by dumping the runner memory so you need to assume that the attacker will be able to get Pull Request write permissions 
https://karimrahal.com/2023/01/05/github-actions-leaking-secrets/

The safest approach is to run all untrusted code in a `pull_request` triggered workflow and pass any needed information to a `workflow_run` privileged workflow but verifying this information to make sure its the expected one.